java/client/src/main/java/glide/managers/ConnectionManager.java

```diff
@@ -38,7 +36,24 @@ public class ConnectionManager {
      */
     public CompletableFuture<Void> connectToRedis(BaseClientConfiguration configuration) {
         ConnectionRequest request = createConnectionRequest(configuration);
-        return channel.connect(request).thenApplyAsync(this::checkGlideRsResponse);
+        return channel
+                .connect(request)
+                .exceptionally(this::exceptionHandler)
+                .thenApplyAsync(this::checkGlideRsResponse);
+    }
+
+    /**
+     * Exception handler for future pipeline.
+     *
+     * @param e An exception thrown in the pipeline before
+     * @return Nothing, it rethrows the exception
+     */
+    private Response exceptionHandler(Throwable e) {
```

Comment:
This looks the same as in CommandManager

Reply:
Yes, but here we always close connection on error. In `CommandManager` we close it on `ClosingError` only.